src/table/internal.tsx

```diff
@@ -119,6 +119,14 @@ const InternalTable = React.forwardRef(
     const [tableWidth, tableMeasureRef] = useContainerQuery<number>(rect => rect.contentBoxWidth);
     const tableRefObject = useRef(null);
 
+    const containerContentWidth = useMemo(() => {
+      if (!containerWidth || !tableRefObject.current) {
+        return null;
+      }
+      const tableStyle = getComputedStyle(tableRefObject.current);
```

Comment:
Here you render the computed value of the previous frame, not your actual state. There could be all sorts of issues, let's avoid this path.

You can only read DOM inside `useEffect` hook

Reply:
P.S. if you need to account paddings, can we just target a different element in our container query 🤔

Reply:
Yes, that is true. The canonical way would be that have a useEffect and a piece of state which is I would expect going to impact performance. I can explain the possible issues in the comments (changing Classic -> VR, changing table's variant) or come up with a better solution. Would you recommend some?

Reply:
Already provided my suggestion: https://github.com/cloudscape-design/components/pull/1650#discussion_r1358239770

Reply:
Ah, see it now. I don't think querying a different element would work. First of all - the paddings are applied to the table and not the container itself and changing that is very tricky. Second, the container width with paddings is also used for setting the empty row width.

Reply:
In any way, the approach in the PR is a no go. We can discuss alternatives, but in a different setting

Reply:
I updated the approach